source/kraken/tests/apply_disruption_test.cpp

```diff
@@ -3305,3 +3305,826 @@ BOOST_AUTO_TEST_CASE(impact_with_timeslots_compare_timeslots_diff_size) {
+                .application_periods(btp("20210101T060000"_dt, "20210101T100000"_dt))
+                .on_rail_section("line:1", "stopAreaC", "stopAreaD", {}, {"route-1"}, *b.data->pt_data)
+                .get_disruption(),
+            *b.data->pt_data, *b.data->meta);
+
+        // Check every relevant fields
+        check_rail_section_impact(b);
+    }
+    /**
+     We receive a disruption with rail_section
+
+                        disruption
+                   -------------------
+                   |        |        |
+     A ------ B ------ C ------ D ------ E ------ F
+                   |        |        |
+                   -------------------
+                  The rail is broken between C and D.
+                  Oh my gosh !
+                  Trains are stopped at C and the pick-up is forbidden
+
+                  -> chaos fields
+                  line               : line:1
+                  start_point        : C
+                  end_point          : D
+                  blocked stop areas : empty
+                  routes             : route:2
+
+        NOTE : No disruptions are added because the route-2 don't have C->D section (This is D->C)
+    */
+    {
+        ed::builder b("20210101", [](ed::builder& b) {
+            b.sa("stopAreaA")("stopA");
+            b.sa("stopAreaB")("stopB");
+            b.sa("stopAreaC")("stopC");
+            b.sa("stopAreaD")("stopD");
+            b.sa("stopAreaE")("stopE");
+            b.sa("stopAreaF")("stopF");
+            b.vj("line:1", "111111", "", true, "vj:A-1")
+                .route("route-1")("stopA", "08:00"_t)("stopB", "08:10"_t)("stopC", "08:20"_t)("stopD", "08:30"_t)(
+                    "stopE", "08:40"_t)("stopF", "08:50"_t);
+            b.vj("line:1", "111111", "", true, "vj:A-2")
+                .route("route-2")("stopF", "08:00"_t)("stopE", "08:10"_t)("stopD", "08:20"_t)("stopC", "08:30"_t)(
+                    "stopB", "08:40"_t)("stopA", "08:50"_t);
+        });
+
+        b.data->meta->production_date = bg::date_period(bg::date(2021, 1, 1), bg::days(7));
+
+        using btp = boost::posix_time::time_period;
+
+        BOOST_REQUIRE_EQUAL(b.data->pt_data->vehicle_journeys.size(), 2);
+
+        // new rail_section disruption
+        navitia::apply_disruption(
+            b.impact(nt::RTLevel::Adapted, "rail_section_on_line1")
+                .severity(nt::disruption::Effect::NO_SERVICE)
+                .application_periods(btp("20210101T060000"_dt, "20210101T100000"_dt))
+                .on_rail_section("line:1", "stopAreaC", "stopAreaD", {}, {"route-2"}, *b.data->pt_data)
+                .get_disruption(),
+            *b.data->pt_data, *b.data->meta);
+
+        // Check every relevant fields
+        BOOST_REQUIRE_EQUAL(b.data->pt_data->lines.size(), 1);
+        BOOST_REQUIRE_EQUAL(b.data->pt_data->routes.size(), 2);
+        BOOST_REQUIRE_EQUAL(b.data->pt_data->vehicle_journeys.size(), 2);
```

Comment:
You can add this test here:
BOOST_REQUIRE_EQUAL(b.data->pt_data->disruption_holder.nb_disruptions(), 0);

Reply:
It is not possible to have **0** for **nb_disurptions**. 
Even if the disruption is rejected, the _builder_helper.cpp_ create an impact... https://github.com/CanalTP/navitia/blob/dev/source/ed/build_helper.cpp#L547
But in the real world we have 0